framework/Optimizers/crossOverOperators/crossovers.py

```diff
@@ -130,12 +130,13 @@ def twoPointsCrossover(parents, parentIndexes,**kwargs):
                               coords={'chromosome': np.arange(int(2*comb(nParents,2))),
                                       'Gene':parents.coords['Gene'].values})
   index = 0
-  for couples in parentIndexes:
+  parentPairs = list(combinations(parents,2))
+  for couples in parentPairs:
     locRangeList = list(range(0,nGenes))
-    index1 = randomUtils.randomIntegers(0, len(locRangeList), caller=None, engine=None)
+    index1 = randomUtils.randomIntegers(0, len(locRangeList)-1, caller=None, engine=None)
```

Comment:
On a second thought: should it be:
 randomUtils.randomIntegers(**1**, len(locRangeList)-1, caller=None, engine=None)